New analysis task added to PWGJE: calculating Jet Charge

JColburn-CERN opened the pull request:
Added a new analysis task AliAnalysisTaskJetCharge which calculates the jet charge distributions for either pp or PbPb.
This has been added such that it can be run on the LEGO trains system.
Written by me, J Samuel Colburn. Originally uploaded on the 28th of October.

alibuild commented:
9a5b65c76874621fe19f63eb2df590c60cc1ae07: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

alibuild commented:
e9a48a2c66aef18c398b9ed6c6edb431fd9d9083: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

raymondEhlers commented on PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeTemplates.h:
```diff
@@ -0,0 +1,144 @@
+
+const Int_t nBranches = 23;
+
+class MyInfo : public fastjet::PseudoJet::UserInfoBase
+{
+  public:
+    Bool_t            kIsGhost;
+    Int_t             fCharge;
+    Int_t             fPartonPdg;
+    MyInfo(Bool_t IsGhost,Int_t Charge,Int_t PartonPdg)
+    {
+    kIsGhost = IsGhost;
+    fCharge = Charge;
+    fPartonPdg = PartonPdg;
+    }
+
+  Int_t getCharge() const{
+    return fCharge;
+  }
+};
```
This is a really generic name to put into the global namespace

raymondEhlers reviewed (changes requested):
Hi,

You need to include these files in the CMakeLists and the LinkDef from them to actually be usable in AliPhysics. I didn't review your code in detail, but made one comment on one of your classes - the name is quite generic and may conflict with others.

Best regards,
Raymond

JColburn-CERN commented:
Hi Raymond,
Thank you for your feedback, This is my first time adding to the repository so I was unaware of those steps so I have now added the files to the cmake list and linkdef.
I have renamed AliAanalysisTaskJetChargeTemplates to AliAnalysisTaskJetChargeFlavourTemplates to make the name less generic. Should I also remane AliAnalysisJetTaskJetCharge to add spesificity?

Thank you for your help
Best regards
Sammy

From: Raymond Ehlers <notifications@github.com>
Reply to: alisw/AliPhysics <reply@reply.github.com>
Date: Friday, 30 October 2020 at 08:47
To: alisw/AliPhysics <AliPhysics@noreply.github.com>
Cc: Jonathan Samuel Colburn <jonathan.samuel.colburn@cern.ch>, Author <author@noreply.github.com>
Subject: Re: [alisw/AliPhysics] New analysis task added to PWGJE: calculating Jet Charge (#15972)


@raymondEhlers requested changes on this pull request.

Hi,

You need to include these files in the CMakeLists and the LinkDef from them to actually be usable in AliPhysics. I didn't review your code in detail, but made one comment on one of your classes - the name is quite generic and may conflict with others.

Best regards,
Raymond

________________________________

In PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeTemplates.h<https://github.com/alisw/AliPhysics/pull/15972#discussion_r514944553>:

> +class MyInfo : public fastjet::PseudoJet::UserInfoBase

+{

+  public:

+    Bool_t            kIsGhost;

+    Int_t             fCharge;

+    Int_t             fPartonPdg;

+    MyInfo(Bool_t IsGhost,Int_t Charge,Int_t PartonPdg)

+    {

+    kIsGhost = IsGhost;

+    fCharge = Charge;

+    fPartonPdg = PartonPdg;

+    }

+

+  Int_t getCharge() const{

+    return fCharge;

+  }

+};

This is a really generic name to put into the global namespace

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/alisw/AliPhysics/pull/15972#pullrequestreview-520466335>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOBWTWJ63MXF2X4VWG7WOCDSNJ4NZANCNFSM4TCUQBUQ>.

alibuild commented:
5c625a05282cdda7b5daa6fa4591b6cf40cda9b6: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

alibuild commented:
2dd97ca17b4179bbc7c9069b374d8269a691a6c5: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

alibuild commented:
24d2be7fbec4a190bacf70a7d0af5b764bba8169: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

alibuild commented:
051de3ef9b1cac3bdc1e6f42996120fe63d7b58b: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

raymondEhlers commented:
Hi Sammy,

Thanks for the fixes. I wasn't actually referring to you task name - it's just fine. My concern was about the class called `MyInfo`, which is why I highlighted it in my review.

Best,
Raymond

alibuild commented:
eb01f39e988f7c59803a5ad3150a18758ec9a705: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

JColburn-CERN commented:
Hi Raymond,
Ah thank you for the clarification that makes more sense,
Also thank you for bringing it to my attention since MyInfo was simply a holdover from an older piece of code and no longer being used so I simply removed it.
Cheers
Sammy

From: Raymond Ehlers <notifications@github.com>
Reply to: alisw/AliPhysics <reply@reply.github.com>
Date: Friday, 30 October 2020 at 15:10
To: alisw/AliPhysics <AliPhysics@noreply.github.com>
Cc: Jonathan Samuel Colburn <jonathan.samuel.colburn@cern.ch>, Author <author@noreply.github.com>
Subject: Re: [alisw/AliPhysics] New analysis task added to PWGJE: calculating Jet Charge (#15972)


Hi Sammy,

Thanks for the fixes. I wasn't actually referring to you task name - it's just fine. My concern was about the class called MyInfo, which is why I highlighted it in my review.

Best,
Raymond

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/alisw/AliPhysics/pull/15972#issuecomment-719609878>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOBWTWP2JJ3HLNTIA5GO2YDSNLJNTANCNFSM4TCUQBUQ>.

raymondEhlers commented:
Hi Sammy,

Thanks for the fixes. You should fix the issues identified by the testing, as well as squash your commits. Also, I don't see a LinkDef entry. Lastly, since you explicitly include fastjet, you need to move to the fastjet section of the CMakeLists. Same for the LinkDef.

Best,
Raymond

alibuild commented:
9d3309813fd8a0aee595d6f78e8485f033b98a73: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

alibuild commented:
a824f51228d3380fa41e7b76dbc26c453685cb85: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

JColburn-CERN commented:
HI Reymond,
Thanks for running the tests. I believe I have fixed the issues that were present in the tests as well as having moved the entry on the CMakeList and added the Linkdef
Thank you for the help,
Sammy

raymondEhlers commented:
+1 Looks good - thanks Sammy

alibuild commented:
a824f51228d3380fa41e7b76dbc26c453685cb85: approved: will be automatically merged on successful tests

alibuild commented:
Error while checking build/AliPhysics/release for a824f51228d3380fa41e7b76dbc26c453685cb85 at 2020-11-01 14:54:
```
sw/BUILD/AliPhysics-latest/log
[ 73%] Built target libPWGDevNanoAOD.rootmap
[ 73%] Built target libPWGGAPHOSTasks.rootmap
[ 73%] Built target libPWGPPMUONdep.rootmap
[ 73%] Built target libPWGUDdiffractive.rootmap
[ 73%] Built target libPWGCFCorrelationsDPhi.rootmap
[ 76%] Built target PWGflowTasks
[ 76%] Built target PWGLFnuclex
[ 76%] Built target PWGDQdielectron
[ 76%] Built target libPWGflowTasks.rootmap
[ 76%] Built target PWGGAGammaConv
[ 77%] Built target PWGCFPLamAnalysisPP
[ 81%] Built target PWGCFfemtoscopy
[ 83%] Built target libPWGLFnuclex.rootmap
[ 83%] Built target PWGLFresonances
[ 83%] Built target PWGJE
[ 83%] Built target PWGHFHFMuFlow
[ 83%] Built target PWGLFspectra
[ 83%] Built target libPWGCFPLamAnalysisPP.rootmap
[ 83%] Built target PWGGAHyperon
[ 83%] Built target libPWGDQdielectron.rootmap
[ 83%] Built target libPWGGAGammaConv.rootmap
[ 83%] Built target PWGCFEfficiencyTask
[ 83%] Built target PWGDQreducedTree
[ 83%] Built target libPWGHFHFMuFlow.rootmap
[ 83%] Built target libPWGJE.rootmap
[ 83%] Built target PWGJEStrangenessInJets
[ 83%] Built target libPWGLFspectra.rootmap
[ 83%] Built target libPWGLFresonances.rootmap
[ 83%] Built target PWGLFspectraJET
[ 83%] Built target libPWGCFfemtoscopy.rootmap
[ 83%] Built target PWGCFFemtoDream
[ 83%] Built target libPWGCFEfficiencyTask.rootmap
[ 83%] Built target PWGCFESE
[ 83%] Built target libPWGGAHyperon.rootmap
[ 83%] Built target libPWGDQreducedTree.rootmap
[ 83%] Built target libPWGLFspectraJET.rootmap
[ 83%] Built target libPWGJEStrangenessInJets.rootmap
[ 83%] Built target libPWGCFESE.rootmap
[ 83%] Built target libPWGCFFemtoDream.rootmap
[ 86%] Built target PWGHFhfe
[ 86%] Built target libPWGHFhfe.rootmap
[ 89%] Built target PWGCFfemtoscopyUser
[ 90%] Built target PWGHFcorrelationHF
[ 90%] Built target libPWGHFcorrelationHF.rootmap
[ 90%] Built target libPWGCFfemtoscopyUser.rootmap
/bin/sh: line 1:  7915 Segmentation fault      (core dumped) LD_LIBRARY_PATH=/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/ROOT/v5-34-30-alice10-63/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/KFParticle/alice-v1.1-2-19/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/AliRoot/v5-09-56-7/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/AliEn-ROOT-Legacy/0.1.3-71/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/Alice-GRID-Utils/0.0.6-3/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/xalienfs/v1.0.14r1-alice3-60/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/SWIG/3.0.12-3/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/Vc/1.4.1-36/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/GEANT4_VMC/v3-6-p6-inclxx-biasing-p2-84/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/vgm/v4-4-84/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/GEANT4/v10.4.2-24/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/GEANT3/v2-7-p2-53/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/GEANT3/v2-7-p2-53/lib64:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/fastjet/v3.2.1_1.024-alice3-58/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/cgal/4.6.3-88/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/MPFR/v3.1.3-4/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/GMP/v6.0.0-4/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/boost/v1.74.0-41/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/bz2/1.0.8-4/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/lzma/v5.2.3-2/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/DPMJET/v3.0.5-alice5-15/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/ROOT/v5-34-30-alice10-63/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/XRootD/v3.3.6-alice2-60/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/Python-modules/1.0-206/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/Python-modules-list/1.0-22/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/Python/v3.6.10-110/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/libffi/v3.2.1-3/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/libffi/v3.2.1-3/lib64:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/sqlite/v3.15.0-3/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/libpng/v1.6.34-43/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/CMake/v3.18.2-30/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/FreeType/v2.10.1-42/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/GSL/v1.16-10/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/alibuild-recipe-tools/0.2.1-2/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/AliEn-Runtime/v2-19-le-11/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/UUID/v2.27.1-1/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/ApMon-CPP/v2.2.8-alice5-5/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/libtirpc/libtirpc-1-1-4-5/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/AliEn-CAs/v1-1/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/OpenSSL/v1.0.2o-8/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/libxml2/v2.9.3-3/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/zlib/v1.2.8-3/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/GCC-Toolchain/v7.3.0-alice2-2/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/GCC-Toolchain/v7.3.0-alice2-2/lib64:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/autotools/v1.6.3-3/lib:/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/defaults-release/v1-3/lib /mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/ROOT/v5-34-30-alice10-63/bin/rootcint -f /mnt/mesos/sandbox/sandbox/aliphysics-release/sw/BUILD/12d174dda605e418225696823b8403d09e088746/AliPhysics/PWGJE/EMCALJetTasks/G__PWGJEEMCALJetTasks.cxx -c -p -DHAVE_FASTJET -D_MODULE_=\"PWGJEEMCALJetTasks\" -DWITHXML -DWITHALIEN -Doverride= -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/UserTasks -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/Tracks -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/fastjet/v3.2.1_1.024-alice3-58/include -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/JETAN/JETAN -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGPP/EVCHAR/FlowVectorCorrections/QnCorrectionsInterface -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGPP/EVCHAR/FlowVectorCorrections/QnCorrections -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGPP/vertexingHF -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGHF/vertexingHF -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGHF/jetsHF -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWG/CaloTrackCorrBase -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWG/Tools/yaml-cpp/include -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWG/Tools -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWG/JETFW -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWG/EMCAL/EMCALtrigger -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWG/EMCAL/EMCALbase -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWG/DevNanoAOD -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/OADB -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/CORRFW -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/KFParticle/alice-v1.1-2-19/include -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/UserTasks -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/ROOT/v5-34-30-alice10-63/include -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/AliEn-Runtime/v2-19-le-11/include -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/AliRoot/v5-09-56-7/include/pythia -I/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/slc7_x86-64/AliRoot/v5-09-56-7/include AliAnalysisTaskDeltaPt.h AliAnalysisTaskEmcalJetSample.h AliAnalysisTaskEmcalJetQA.h AliAnalysisTaskEmcalJetSpectraQA.h AliAnalysisTaskEmcalJetTagger.h AliAnalysisTaskPWGJEQA.h AliAnalysisTaskLocalRho.h AliAnalysisTaskRhoAverage.h AliAnalysisTaskRhoBase.h AliAnalysisTaskRho.h AliAnalysisTaskRhoFlow.h AliAnalysisTaskRhoMassBase.h AliAnalysisTaskRhoMass.h AliAnalysisTaskRhoMassSparse.h AliAnalysisTaskRhoSparse.h AliAnalysisTaskJetUE.h AliAnalysisTaskRhoBaseDev.h AliAnalysisTaskRhoDev.h AliAnalysisTaskRhoTransDev.h AliAnalysisTaskScale.h AliEmcalJetByJetCorrection.h AliEmcalJetTaggerTaskFast.h AliEmcalPicoTrackInGridMaker.h AliJetConstituentTagCopier.h AliJetEmbeddingFromGenTask.h AliJetEmbeddingTask.h AliJetFastSimulation.h AliJetModelBaseTask.h AliJetModelCopyTracks.h AliJetModelMergeBranches.h AliJetRandomizerTask.h AliJetResponseMaker.h AliJetTriggerSelectionTask.h AliNanoAODArrayMaker.h AliPWGJETrainHelpers.h AliAnalysisTaskEmcalTriggerBase.h AliAnalysisTaskEmcalTriggerPosition.h AliAnalysisTaskPtEMCalTrigger.h AliAnalysisTaskPtEMCalTriggerV1.h AliCutValueRange.h AliEmcalTriggerOfflineSelection.h AliEMCalTriggerExtraCuts.h AliEMCalTriggerAnaTriggerDecision.h AliEMCalTriggerAnaTriggerDecisionConfig.h AliEMCalTriggerBinningComponent.h AliEMCalTriggerBinningFactory.h AliEMCalTriggerClusterAnalysisComponent.h AliEMCalTriggerEventCounterAnalysisComponent.h AliEMCalTriggerEventData.h AliEMCalTriggerEventSelection.h AliEMCalTriggerKineCuts.h AliEMCalTriggerMCJetAnalysisComponent.h AliEMCalTriggerMCParticleAnalysisComponent.h AliEMCalTriggerPatchAnalysisComponent.h AliEMCalTriggerRecJetAnalysisComponent.h AliEMCalTriggerRecTrackAnalysisComponent.h AliEMCalTriggerTaskGroup.h AliEMCalTriggerTracksAnalysisComponent.h AliEMCalTriggerWeightHandler.h AliEMCalTriggerAnaTriggerClass.h AliEMCalTriggerAnaClassManager.h AliParticleMap.h AliReducedHighPtEventCreator.h AliReducedEmcalCluster.h AliReducedPatchInfo.h AliReducedPatchContainer.h AliReducedMCHeader.h AliReducedGeneratedParticle.h AliReducedReconstructedTrack.h AliReducedHighPtEvent.h AliReducedJetEvent.h AliReducedJetInfo.h AliReducedJetParticle.h AliReducedJetConstituent.h AliReducedMatchedTrack.h AliAnalysisTaskTrackDCA.h AliEmcalAnalysisFactory.h AliAnalysisTaskChargedParticlesRef.h AliAnalysisTaskChargedParticlesRefMC.h AliAnalysisTaskChargedParticlesMCTriggerMimic.h AliAnalysisTaskEmcalTriggerMultiplicity.h AliAnalysisTaskEmcalClustersRef.h AliAnalysisTaskEmcalClusterMatched.h AliAnalysisTaskEmcalPatchesRef.h AliAnalysisTaskEmcalOnlinePatchesRef.h AliAnalysisTaskEmcalOfflinePatchesRef.h AliAnalysisTaskEmcalRecalcPatchesRef.h AliAnalysisTaskEmcalTriggerCorrelation.h AliAnalysisTaskEventSelectionRef.h AliAnalysisTaskEventFilter.h AliAnalysisTaskEtaPhiEfficiency.h AliAnalysisTaskParticleInJet.h AliAnalysisTaskTrackDensity.h AliAnalysisTaskTrackDensityData.h AliEmcalTriggerEmulation.h AliAnalysisTaskEmcalTriggerEmulation.h AliAnalysisTaskEGAMonitor.h AliAnalysisTaskEmcalPatchMasking.h AliAnalysisTaskCountITStracks.h AliAnalysisTaskEmcalTriggerJets.h AliAnalysisTaskEmcalTriggerJetsIDcorr.h AliAnalysisTaskEmcalJetConstituentQA.h AliAnalysisEmcalTriggerSelectionHelper.h AliAnalysisTaskEmcalJetEnergyScale.h AliAnalysisTaskEmcalJetEnergySpectrum.h AliAnalysisTaskEmcalJetSpectrumSDPart.h AliAnalysisTaskEmcalResponseOutliers.h AliAnalysisTaskEmcalOutliersGen.h AliAnalysisTaskEmcalPatchMultCorr.h AliAnalysisTaskEmcalClustersInJets.h AliAnalysisTaskEmcalTriggerCorrelationMC.h AliAnalysisTaskEmcalHighEClusterTree.h AliAnalysisTaskEmcalEG1Correlation.h AliAnalysisTaskEmcalTriggerBackground.h AliAnalysisTaskEmcalQoverPtShift.h AliAnalysisTaskEmcalJetIterativeDeclustering.h AliAnalysisTaskEmcalFastorMultiplicity.h AliAnalysisTaskEmcalTriggerSelectionTest.h AliEmcalTRDTrackingTask.h AliAnalysisTaskJetUEStudies.h AliAnalysisTaskBackFlucRandomCone.h AliAnalysisTaskChargedJetsPA.h AliAnalysisTaskChargedJetsHadronCF.h AliAnalysisTaskChargedJetsHadronToy.h Cumulants.h AliAnalysisTaskCLQA.h AliAnalysisTaskDcalDijetPerf.h AliAnalysisTaskDeltaPtJEmb.h AliAnalysisTaskDijetHadron.h AliAnalysisTaskParticleRandomizer.h AliAnalysisTaskEmcalBadCells.h AliAnalysisTaskEmcalDiJetAna.h AliAnalysisTaskEmcalDiJetBase.h AliAnalysisTaskEmcalDiJetResponse.h AliAnalysisTaskEmcalDijetImbalance.h AliAnalysisTaskEmcalVsPhos.h AliAnalysisTaskEmcalJetPerformance.h AliAnalysisTaskGenerateThermalBackgroundMC.h AliAnalysisTaskEmcalHighMultTrigger.h AliAnalysisTaskEmcalHJetMass.h AliAnalysisTaskIDFragmentationFunction.h AliAnalysisTaskEmcalJetCorrection.h AliAnalysisTaskEmcalJetHadCorQA.h AliAnalysisTaskEmcalJetHadEPpid.h AliAnalysisTaskEmcalJetHUtils.h AliAnalysisTaskEmcalJetHPerformance.h AliAnalysisTaskEmcalJetHCorrelations.h AliAnalysisTaskEmcalJetMassBkg.h AliAnalysisTaskEmcalJetMass.h AliAnalysisTaskEmcalJetMassResponse.h AliAnalysisTaskEmcalJetMassStructure.h AliAnalysisTaskEmcalJetPatchTriggerQA.h AliAnalysisTaskEmcalJetSparseMaker.h AliAnalysisTaskEmcalJetSpectra.h AliAnalysisTaskEmcalJetSpectra8TeVTriggerQA.h AliAnalysisTaskEmcalJetSpectraMECpA.h AliAnalysisTaskEmcalJetTriggerMatcher.h AliAnalysisTaskEmcalJetTriggerQA.h AliAnalysisTaskEmcalJetv2QA.h AliAnalysisTaskEmcalRun2QA.h AliAnalysisTaskEmcalHfeTagging.h AliAnalysisTaskFakeJets.h AliAnalysisTaskEmcalMissingEnergy.h AliAnalysisTaskEmcalNeutralJets.h AliAnalysisTaskEmcalTriggerPatchJetMatch.h AliAnalysisTaskEmcalTmpSparseMaker.h AliAnalysisTaskEmcalTriggerInfoQA.h AliAnalysisTaskEmcalTriggerTreeWriter.h AliAnalysisTaskEventExtractor.h AliAnalysisTaskHJetEmbed.h AliAnalysisTaskJetJTJT.h AliAnalysisTaskJetMassResponseDet.h AliAnalysisTaskJetMatching.h AliAnalysisTaskJetsEvshape.h AliAnalysisTaskJetShapeBase.h AliAnalysisTaskJetShapeConst.h AliAnalysisTaskJetShapeDeriv.h AliAnalysisTaskJetShapeGR.h AliAnalysisTaskJetV2.h AliAnalysisTaskJetV3.h AliAnalysisTaskJetVn.h AliAnalysisTaskPID.h AliAnalysisTaskPIDV0base.h AliAnalysisTaskPSHFE.h AliAnalysisTaskPrepareInputForEmbedding.h AliAnalysisTaskRhoMassScale.h AliAnalysisTaskSOH.h AliAnalysisTaskTriggerRejection.h AliAnalysisTaskV0sInJetsEmcal.h AliEmcalPicoTrackFromJetMaker.h AliAnalysisTaskPi0Hadron.h AliAnalysisTaskEmcalJetTree.h AliJetEmbeddingSelRhoTask.h AliTrackContainerV0.h AliTrackContainerToyModel.h AliMCParticleContainerToyModel.h AliFJWrapper.h AliEmcalJetUtility.h AliEmcalJetUtilityGenSubtractor.h AliEmcalJetUtilityConstSubtractor.h AliEmcalJetUtilityEventSubtractor.h AliEmcalJetUtilitySoftDrop.h AliEmcalJetTask.h AliEmcalJetFinder.h AliJetEmbeddingFromAODTask.h AliJetEmbeddingFromPYTHIATask.h AliJetShape.h AliLundPlaneHelper.h AliAnalysisTaskJetCharge.h AliAnalysisTaskJetChargeFlavourTemplates.h AliAnalysisTaskEmcalQGTagging.h AliAnalysisTaskNewJetSubstructure.h AliAnalysisTaskCheckDeadCone.h AliAnalysisTaskJetPull.h AliAnalysisTaskLundPlane.h AliAnalysisTaskJetDynamicalGrooming.h AliAnalysisTaskJetHardestKt.h AliAnalysisTaskHardestBranch.h AliAnalysisTaskEmcalJetShapesMC.h AliAnalysisTaskEmcalJetShapeExtra.h AliAnalysisTaskFullpAJets.h AliAnalysisTaskFullppJet.h AliAnalysisTaskHJetDphi.h AliAnalysisTaskHJetSpectra.h AliAnalysisTaskJetExtractor.h AliAnalysisTaskJetPP.h AliAnalysisTaskSVtaskMCFilter.h AliAnalysisTaskEA.h AliAnalysisEmcalJetHelperEA.h AliAnalysisTaskPythiaBranchEA.h AliAnalysisTaskSubJetFraction.h AliAnalysisTaskRecoilJetYield.h AliAnalysisTaskRecursiveSoftDrop.h AliAnalysisTaskJetSubstructure.h AliAnalysisTaskSoftDrop.h AliAnalysisTaskSoftDropResponse.h AliAnalysisTaskSDKL.h AliAnalysisTaskSDKLResponse.h AliAnalysisTaskEmcalSubjet.h AliAnalysisTaskJetCoreEmcal.h AliAnalysisTaskEmcalJetCDF.h AliAnalysisTaskPtEfficiencyJets.h AliHighPtReconstructionEfficiency.h AliAnalysisTaskTracksInJet.h AliAnalysisTaskEmcalJetSubstructureTree.h AliAnalysisEmcalSoftdropHelper.h AliAnalysisTaskEmcalSoftDropData.h AliEMCalTriggerAnaHelper.h PWGJEEMCALJetTasksLinkDef.h
gmake[2]: *** [PWGJE/EMCALJetTasks/G__PWGJEEMCALJetTasks.cxx] Error 139
gmake[2]: *** Deleting file `PWGJE/EMCALJetTasks/G__PWGJEEMCALJetTasks.cxx'
gmake[1]: *** [PWGJE/EMCALJetTasks/CMakeFiles/PWGJEEMCALJetTasks-object.dir/all] Error 2
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/alisw/AliPhysics/15972/a824f51228d3380fa41e7b76dbc26c453685cb85/build_AliPhysics_release/pretty.html).

alibuild commented:
Error while checking build/AliPhysics/root6 for a824f51228d3380fa41e7b76dbc26c453685cb85 at 2020-11-01 15:06:
```
sw/BUILD/AliPhysics-latest/log
   Bool_t bTPCqVector(kFALSE), bTPCaSideqVector(kFALSE), bTPCcSideqVector(kFALSE), bV0AqVector(kFALSE), bV0CqVector(kFALSE), bV0qVector(kFALSE),bSPDqVector(kFALSE), bFMDAqVector(kFALSE), bFMDCqVector(kFALSE), bZDCAqVector(kFALSE), bZDCCqVector(kFALSE);
                                                                                                                                                                                                                                       ^~~~~~~~~~~~
In file included from input_line_11:226:
/mnt/mesos/sandbox/sandbox/aliphysics-root6/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.h:92:51: error: invalid use of non-static data member 'nBranches'
  Double_t                            fTreeBranch[nBranches];
                                                  ^~~~~~~~~
In file included from input_line_11:227:
/mnt/mesos/sandbox/sandbox/aliphysics-root6/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeFlavourTemplates.h:118:51: error: invalid use of non-static data member 'nBranches'
  Double_t                            fTreeBranch[nBranches];
                                                  ^~~~~~~~~
[ 34%] Building CXX object PWGGA/PHOSTasks/CMakeFiles/PWGGAPHOSTasks-object.dir/PHOS_PbPb/AliAnalysisTaskPi0Flow.cxx.o
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/alisw/AliPhysics/15972/a824f51228d3380fa41e7b76dbc26c453685cb85/build_AliPhysics_root6/pretty.html).

alibuild commented:
7b7c4295563446579fa171d409f3edd9497c364d: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

raymondEhlers commented:
+1

alibuild commented:
7b7c4295563446579fa171d409f3edd9497c364d: approved: will be automatically merged on successful tests

alibuild commented:
Error while checking build/AliPhysics/release for 7b7c4295563446579fa171d409f3edd9497c364d at 2020-11-03 11:18:
```
sw/BUILD/AliPhysics-latest/log
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx:87:1: warning:   when initialized here [-Wreorder]
 AliAnalysisTaskJetCharge::AliAnalysisTaskJetCharge(const char *name) :
 ^~~~~~~~~~~~~~~~~~~~~~~~
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx:112:19: error: 'nBranches' was not declared in this scope
   for(Int_t i=0;i<nBranches;i++){
                   ^~~~~~~~~
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx:112:19: note: suggested alternative: 'GetBranches'
--
                   ^~~~~~~~~
                   GetBranches
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx: In member function 'virtual void AliAnalysisTaskJetCharge::UserCreateOutputObjects()':
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx:144:43: error: 'nBranches' was not declared in this scope
   TString *fTreeBranchName = new TString [nBranches];
                                           ^~~~~~~~~
In file included from /mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeFlavourTemplates.cxx:48:0:
--
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeFlavourTemplates.cxx:115:1: warning:   when initialized here [-Wreorder]
 AliAnalysisTaskJetChargeFlavourTemplates::AliAnalysisTaskJetChargeFlavourTemplates(const char *name) :
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeFlavourTemplates.cxx:166:19: error: 'nBranches' was not declared in this scope
   for(Int_t i=0;i<nBranches;i++){
                   ^~~~~~~~~
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx:144:43: note: suggested alternative: 'GetBranches'
--
                   ^~~~~~~~~
                   GetBranches
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeFlavourTemplates.cxx: In member function 'virtual void AliAnalysisTaskJetChargeFlavourTemplates::UserCreateOutputObjects()':
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeFlavourTemplates.cxx:195:43: error: 'nBranches' was not declared in this scope
   TString *fTreeBranchName = new TString [nBranches];
                                           ^~~~~~~~~
/mnt/mesos/sandbox/sandbox/aliphysics-release/sw/SOURCES/AliPhysics/build_AliPhysics_release/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetChargeFlavourTemplates.cxx:195:43: note: suggested alternative: 'GetBranches'
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/alisw/AliPhysics/15972/7b7c4295563446579fa171d409f3edd9497c364d/build_AliPhysics_release/pretty.html).

alibuild commented:
Error while checking build/AliPhysics/root6 for 7b7c4295563446579fa171d409f3edd9497c364d at 2020-11-03 11:22:
```
sw/BUILD/AliPhysics-latest/log
/mnt/mesos/sandbox/sandbox/aliphysics-root6/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx:87:1: warning:   when initialized here [-Wreorder]
 AliAnalysisTaskJetCharge::AliAnalysisTaskJetCharge(const char *name) :
 ^~~~~~~~~~~~~~~~~~~~~~~~
/mnt/mesos/sandbox/sandbox/aliphysics-root6/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx:112:19: error: 'nBranches' was not declared in this scope
   for(Int_t i=0;i<nBranches;i++){
                   ^~~~~~~~~
/mnt/mesos/sandbox/sandbox/aliphysics-root6/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx:112:19: note: suggested alternative: 'GetBranches'
--
                   ^~~~~~~~~
                   GetBranches
/mnt/mesos/sandbox/sandbox/aliphysics-root6/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx: In member function 'virtual void AliAnalysisTaskJetCharge::UserCreateOutputObjects()':
/mnt/mesos/sandbox/sandbox/aliphysics-root6/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx:144:43: error: 'nBranches' was not declared in this scope
   TString *fTreeBranchName = new TString [nBranches];
                                           ^~~~~~~~~
/mnt/mesos/sandbox/sandbox/aliphysics-root6/sw/SOURCES/AliPhysics/build_AliPhysics_root6/0/PWGJE/EMCALJetTasks/AliAnalysisTaskJetCharge.cxx:144:43: note: suggested alternative: 'GetBranches'
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/alisw/AliPhysics/15972/7b7c4295563446579fa171d409f3edd9497c364d/build_AliPhysics_root6/pretty.html).

alibuild commented:
6ed27bec81961e4605441462f5984e005c677ef0: approval required: 1 of @vkucera (Vit Kucera), @rhaake (Rudiger Haake), @adriansev (Adrian Sevcenco), @qgp (Jochen Klein), @matplo (Mateusz Ploskon), @jgrosseo (Jan Fiete Grosse-Oetringhaus), @shahor02 (Ruben Shahoyan), @davidrohr (David Rohr), @dcaffarr (Davide Caffarri), @mfasDa (Markus Fasel), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @chiarazampolli (Chiara Zampolli), @lcunquei (Leticia Cunqueiro Mendez)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/15972) and comment at the bottom of the page.**_

raymondEhlers commented:
+1

alibuild commented:
6ed27bec81961e4605441462f5984e005c677ef0: approved: will be automatically merged on successful tests
